### ISSUE TITLE
All Paths in SHORTEST_PATH and GRAPH_SHORTEST_PATH

### ISSUE THREAD

**Issue by mchacki** [arangodb/arangodb#1519]
Both AQL functions SHORTEST_PATH and GRAPH_SHORTEST_PATH only return one shortest path even if there are multiple candidates. It should be possible to use an option to return all found shortest paths in both cases.

**Comment by fceller** [comment]
@mchacki has this been fixed in 2.7.0?

**Comment by mchacki** [comment]
No this is a feature request not a bug. Includes some major implementation.

**Comment by judioo** [comment]
Any idea when this will be started on. Pretty critical  requirement for myself 👍

**Comment by judioo** [comment]
What would _really_ be useful is to be able to name an intermediary vertice /edge that the path must go through

```
FOR s IN GRAPH_SHORTEST_PATH (
    'careerPath', 
    "role/1738940225",
    "role/1174313602", 
    {
        direction: "outbound", 
        includeData: true,
        intermediaryVertex: ["role/23879"],
        weight: "weight"
    }
)
RETURN s
```

**Comment by vmaasalo** [comment]
@judioo about intermediary vertex: Would finding the path in two segments work – first from A to B, then B to C? (B = intermediary vertex; A&C the endpoints)

**Comment by maxkernbach** [comment]
The AQL function SHORTEST_PATH still returns only shortest path. However it is possible to receive all shortest paths by combining two separate queries (see [Multiple Path Search](https://docs.arangodb.com/latest/AQL/Examples/MultiplePaths.html) in the documentation).

**Comment by maxkernbach** [comment]
Closing this issue as fixed since the **k Shortest Paths** feature is available from 3.5 RC 1.

> This type of query is supposed to find the first k paths in order of length (or weight) between two given documents, startVertex and targetVertex in your graph.

Please check the links below for further details:

- [k Shortest Paths tutorial](https://www.arangodb.com/arangodb-training-center/graphs/k-shortest-paths-queries-in-aql/)
- [k Shortest Paths documentation
](https://www.arangodb.com/docs/devel/aql/graphs-kshortest-paths.html)

The latest Release Candidate of version 3.5 can be found in the [download section](https://www.arangodb.com/download-technical-preview/).
